common/domain/replication_queue_test.go

```diff
@@ -619,3 +710,28 @@ func TestPurgeAckedMessages(t *testing.T) {
 		})
 	}
 }
+
+func TestReplicationQueueImpl_purgeProcessor(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockQueue := persistence.NewMockQueueManager(ctrl)
+	rq := NewReplicationQueue(mockQueue, "testCluster", nil, nil).(*replicationQueueImpl)
+	atomic.StoreInt32(&rq.status, common.DaemonStatusStarted)
+
+	done := make(chan bool)
+	mockQueue.EXPECT().GetAckLevels(gomock.Any()).Return(map[string]int64{}, nil).AnyTimes()
+	mockQueue.EXPECT().DeleteMessagesBefore(gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+
+	go func() {
+		rq.purgeProcessor()
+		close(done)
+	}()
+
+	time.Sleep(1 * time.Nanosecond)
```

Comment:
sleeping should be unnecessary here, and it doesn't really do anything for the purposes of this test - it'll effectively always take the `<-done` route because the ticker won't fire before `rq.Stop()` is called.